grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -908,23 +905,20 @@ class OrderService {
                 'estimatedReadyDate',
                 'actualReadyDate',
                 'budgetCode'
-            ]
+        ]
+
+        char dataSeparator =  CSVUtils.getSeparator(text, fieldKeys.size())
+
+        try {
+            def settings = [skipLines: 1, separatorChar: dataSeparator]
+            def csvMapReader = new CSVMapReader(new StringReader(text), settings)
+            csvMapReader.fieldKeys = fieldKeys
             orderItems = csvMapReader.toList()
 
         } catch (Exception e) {
             throw new RuntimeException("Error parsing order item CSV: " + e.message, e)
         }
 
-        // FIXME it's not entirely clear why we reformat strings here like this
-        orderItems.each { orderItem ->
-            if (orderItem.unitOfMeasure) {
-                String[] uomParts = orderItem.unitOfMeasure.split("/")
-                def quantityUom = CSVUtils.parseNumber(uomParts[1], "unitOfMeasure")
-                orderItem.unitOfMeasure = "${uomParts[0]}/${quantityUom}"
-            }
-            orderItem.unitPrice = orderItem.unitPrice ? CSVUtils.parseNumber(orderItem.unitPrice, "unitPrice").setScale(4, RoundingMode.FLOOR).toString() : ''
-        }
-
```

Comment:
The FIXME comment above is right, it is not clear why such an operation was required but this reformatting of **unitPrice** and **unitOfMeasure** has introduced a bug when trying to import OrderItems in different locales.

The reason for the bug is that by using `CSVUtils.parseNumber(...)` it parses the number to english format with a comma which does not work in other locales than english and that caused the imported prices to be wrongly formatted in other locales.

While discussing this issue we came to the conclusion that we won't support a cross language export/import.
We do not expect a user to **export** a file in **english** locale the **switch** to **french** locale and try to import that file.